src/HeaderAddPlugin.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Adds headers to the request only if the request does not already own it.
```

Comment:
it currently is true. i guess it depends on the header whether appending makes sense or not, but the code below only adds the header if its not there yet. for many, that does make sense.